method and interface with the same name

### The following program `sample.go` triggers an unexpected result

```go
package main

type I2 interface {
    I2() string
}

type I interface {
    I2
}

type S struct{}

func (*S) I2() string { return "foo" }

func main() {
    var i I
    _, ok := i.(*S)
    println(ok)
}
```


### Expected result

```console
$ go run ./sample.go
false
```


### Got

```console
$ yaegi ./sample.go
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: /home/marc/yt/issue-1419-2.go:17:11: CFG post-order panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x7da514]

goroutine 1 [running]:
github.com/traefik/yaegi/interp.(*Interpreter).cfg.func2.1()
        /home/marc/go/src/github.com/traefik/yaegi/interp/cfg.go:536 +0x78
panic({0xdd04e0, 0x157bad0})
        /home/marc/sdk/go1.19/src/runtime/panic.go:890 +0x262
github.com/traefik/yaegi/interp.isInterfaceSrc(0xc0001868b0?)
        /home/marc/go/src/github.com/traefik/yaegi/interp/type.go:2270 +0x14
github.com/traefik/yaegi/interp.isInterface(0xc000200140?)
        /home/marc/go/src/github.com/traefik/yaegi/interp/type.go:2278 +0x25
github.com/traefik/yaegi/interp.(*itype).equals(0x0, 0xc0002d3200)
        /home/marc/go/src/github.com/traefik/yaegi/interp/type.go:1441 +0x3b
github.com/traefik/yaegi/interp.typecheck.typeAssertionExpr({0xc000187200?}, 0xc00043aa20, 0xc00043b7a0)
        /home/marc/go/src/github.com/traefik/yaegi/interp/typecheck.go:643 +0xb45
github.com/traefik/yaegi/interp.(*Interpreter).cfg.func2(0xc00043a900)
        /home/marc/go/src/github.com/traefik/yaegi/interp/cfg.go:2133 +0x7a3f
github.com/traefik/yaegi/interp.(*node).Walk(0xc00043a900, 0xc000187990, 0xc0001879d8)
        /home/marc/go/src/github.com/traefik/yaegi/interp/interp.go:291 +0xad
github.com/traefik/yaegi/interp.(*node).Walk(0xc00043a5a0, 0xc000187990, 0xc0001879d8)
        /home/marc/go/src/github.com/traefik/yaegi/interp/interp.go:288 +0x75
github.com/traefik/yaegi/interp.(*node).Walk(0xc000439e60, 0xc000187990, 0xc0001879d8)
        /home/marc/go/src/github.com/traefik/yaegi/interp/interp.go:288 +0x75
github.com/traefik/yaegi/interp.(*node).Walk(0xc0004397a0, 0xc000187990, 0xc0001879d8)
        /home/marc/go/src/github.com/traefik/yaegi/interp/interp.go:288 +0x75
github.com/traefik/yaegi/interp.(*node).Walk(0xc0002ea7e0, 0xc000187990, 0xc0001879d8)
        /home/marc/go/src/github.com/traefik/yaegi/interp/interp.go:288 +0x75
github.com/traefik/yaegi/interp.(*Interpreter).cfg(0xc0002cc000, 0xc0002ea7e0, 0xc0002d42d0, {0xc0002119a0, 0x4}, {0xc0002119a0, 0x4})
        /home/marc/go/src/github.com/traefik/yaegi/interp/cfg.go:62 +0x2b4
github.com/traefik/yaegi/interp.(*Interpreter).CompileAST(0xc0002cc000, {0x1097cc8?, 0xc000218200?})
        /home/marc/go/src/github.com/traefik/yaegi/interp/program.go:92 +0x1ad
github.com/traefik/yaegi/interp.(*Interpreter).compileSrc(0xc0002cc000, {0xc00029e840?, 0x0?}, {0x7ffe69395970?, 0xc00029e840?}, 0xbc?)
        /home/marc/go/src/github.com/traefik/yaegi/interp/program.go:59 +0xb8
github.com/traefik/yaegi/interp.(*Interpreter).eval(0xc0002cc000, {0xc00029e840?, 0xbc?}, {0x7ffe69395970?, 0xc00029e6c0?}, 0xbc?)
        /home/marc/go/src/github.com/traefik/yaegi/interp/interp.go:552 +0x28
github.com/traefik/yaegi/interp.(*Interpreter).EvalPath(0xc0002cc000, {0x7ffe69395970, 0x1d})
        /home/marc/go/src/github.com/traefik/yaegi/interp/interp.go:510 +0xab
main.runFile(0x7ffe69395970?, {0x7ffe69395970, 0x1d}, 0x0)
        /home/marc/go/src/github.com/traefik/yaegi/cmd/yaegi/run.go:153 +0xee
main.run({0xc000036050?, 0x1, 0x1})
        /home/marc/go/src/github.com/traefik/yaegi/cmd/yaegi/run.go:116 +0xbec
main.main()
        /home/marc/go/src/github.com/traefik/yaegi/cmd/yaegi/yaegi.go:144 +0x2cb
exit status 2
```


### Yaegi Version

v0.14.2

### Additional Notes

This bug is related to #1419 once #1452 is applied.

--- a/_test/issue-1454.go
+++ b/_test/issue-1454.go
@@ -1,0 +1,19 @@
+package main
+
+type I2 interface {
+	I2() string
+}
+
+type I interface {
+	I2
+}
+
+type S struct{}
+
+func (*S) I2() string { return "foo" }
+
+func main() {
+	var i I
+	_, ok := i.(*S)
+	println(ok)
+}

--- a/_test/issue-1454.go
+++ b/_test/issue-1454.go
@@ -17,3 +17,6 @@ func main() {
 	_, ok := i.(*S)
 	println(ok)
 }
+
+// Output:
+// false

--- a/interp/cfg.go
+++ b/interp/cfg.go
@@ -1060,7 +1060,7 @@ func (interp *Interpreter) cfg(root *node, sc *scope, importPath, pkgName string
 				}
 
 				n.gen = c0.sym.builtin
-				c0.typ = &itype{cat: builtinT}
+				c0.typ = &itype{cat: builtinT, name: bname}
 				if n.typ, err = nodeType(interp, sc, n); err != nil {
 					return
 				}

--- a/interp/type.go
+++ b/interp/type.go
@@ -1249,8 +1249,13 @@ func (t *itype) numOut() int {
 		if t.rtype.Kind() == reflect.Func {
 			return t.rtype.NumOut()
 		}
+	case builtinT:
+		switch t.name {
+		case "append", "cap", "complex", "copy", "imag", "len", "make", "new", "real", "recover":
+			return 1
+		}
 	}
-	return 1
+	return 0
 }
 
 func (t *itype) out(i int) *itype {

--- a/interp/typecheck.go
+++ b/interp/typecheck.go
@@ -625,6 +625,11 @@ func (check typecheck) typeAssertionExpr(n *node, typ *itype) error {
 			return n.cfgErrorf("impossible type assertion: %s does not implement %s as %q method has a pointer receiver", typ.id(), n.typ.id(), name)
 		}
 
+		if im.cat != funcT || tm.cat != funcT {
+			// It only makes sense to compare in/out parameter types if both types are functions.
+			continue
+		}
+
 		err := n.cfgErrorf("impossible type assertion: %s does not implement %s", typ.id(), n.typ.id())
 		if im.numIn() != tm.numIn() || im.numOut() != tm.numOut() {
 			return err

